packages/compile-solidity/legacy/shims.js

```diff
@@ -1,13 +1,15 @@
-function shimOutput({ contracts: list, sourceIndexes, compilerInfo }) {
+const { shims } = require("@truffle/compile-common");
+
+function shimOutput({ contracts: list, sourceIndexes, compiler }) {
```

Comment:
I believe this shim is only for `compile-solidity/legacy`. I tried to put everything legacy-related quarantined off in the `legacy` folder so it can just be easily deleted later.

Reply:
Oh, we should definitely get all the shims in one place. No need to cordon off the implementation - inside `compile-solidity/legacy` we can import from elsewhere, no?

Reply:
Oh I see, you're already importing the shims for `forContract`. So I think all that's left is to get rid of `shimContract`, etc. from this file, since it doesn't appear to be used anymore.

Reply:
Ah right, it looks like that function is unused.